react/src/components/AgentList.tsx

```diff
@@ -0,0 +1,816 @@
+    query: agentListQuery,
+    defaultVariables: {
+      page: 1,
+      pageSize: 50,
+      filter,
+      order,
+    },
+  });
+
+  const { agent_list } = useLazyLoadQuery<AgentListQuery>(
+    agentListQuery,
+    paginationStates.variables,
+    paginationStates.refreshedQueryOptions,
+  );
+
+  const columns: ColumnsType<Agent> = [
+    {
+      title: '#',
+      fixed: 'left',
+      render: (id, record, index) => {
+        ++index;
+        return index;
+      },
+      showSorterTooltip: false,
+      rowScope: 'row',
+    },
+    {
+      title: t('agent.Endpoint'),
+      key: 'id',
+      dataIndex: 'id',
+      fixed: 'left',
+      render: (value, record) => {
+        return (
+          <Flex direction="column" align="start">
+            <Typography.Text>{value}</Typography.Text>
+            <Typography.Text type="secondary">{record?.addr}</Typography.Text>
+          </Flex>
+        );
+      },
+      sorter: (a, b) => (a?.id && b?.id ? a.id.localeCompare(b.id) : 0),
+    },
+    {
+      title: t('agent.Region'),
+      key: 'region',
+      dataIndex: 'region',
+      render: (value) => {
+        const platformData: {
+          [key: string]: { color: string; icon: string };
+        } = {
+          aws: { color: 'orange', icon: 'aws' },
+          amazon: { color: 'orange', icon: 'aws' },
+          azure: { color: 'blue', icon: 'azure' },
+          gcp: { color: 'lightblue', icon: 'gcp' },
+          google: { color: 'lightblue', icon: 'gcp' },
+          nbp: { color: 'green', icon: 'nbp' },
+          naver: { color: 'green', icon: 'nbp' },
+          openstack: { color: 'red', icon: 'openstack' },
+          dgx: { color: 'green', icon: 'local' },
+          local: { color: 'yellow', icon: 'local' },
+        };
+
+        const regionData = _.split(value, '/');
+        const platform = regionData?.[0];
+        const location = regionData?.length > 1 ? regionData[1] : '';
+        const { color, icon } = platformData[platform] || {
+          color: 'yellow',
+          icon: 'local',
+        };
+        return (
+          <Flex>
+            <img
+              alt={value}
+              src={`/resources/icons/${icon}.png`}
+              style={{
+                width: '32px',
+                height: '32px',
+                filter: isDarkMode && icon === 'local' ? 'invert(1)' : '',
+              }}
+            />
+            {location !== '' ? (
+              <DoubleTag
+                values={[
+                  { label: location, color: color },
+                  { label: platform, color: color },
+                ]}
+              />
+            ) : (
+              <Tag color={color}>{platform}</Tag>
+            )}
+          </Flex>
+        );
+      },
+    },
+    {
+      title: t('agent.Architecture'),
+      key: 'architecture',
+      dataIndex: 'architecture',
+      sorter: (a, b) =>
+        (a?.architecture ?? '').localeCompare(b?.architecture ?? ''),
```

Comment:
Let's add the following function to `/helper/index.ts` and use it in the same way as the Control panel.
```
export const localeCompare = (a?: string | null, b?: string | null) => {
  if (a === null || a === undefined) return -1;
  if (b === null || b === undefined) return 1;
  return a.localeCompare(b);
};
```